### PR TITLE
Fix `to_torchscript()` causing false positive deprecation warnings

### LINKED ISSUE #10092: `LightningModule.to_torchscript()` causes false positive deprecation warnings
### To Reproduce

```py
import torch

from pytorch_lightning import LightningModule


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)


def run():
    model = BoringModel()
    scripted = model.to_torchscript()
    print(scripted)


if __name__ == "__main__":
    run()

```

```
/Users/adrian/.conda/envs/lightning/lib/python3.8/site-packages/torch/jit/_recursive.py:614: LightningDeprecationWarning: The `LightningModule.loaded_optimizer_states_dict` property is deprecated in v1.4 and will be removed in v1.6.
  item = getattr(mod, name, None)
/Users/adrian/.conda/envs/lightning/lib/python3.8/site-packages/torch/jit/_recursive.py:614: LightningDeprecationWarning: The `LightningModule.model_size` property was deprecated in v1.5 and will be removed in v1.7. Please use the `pytorch_lightning.utilities.memory.get_model_size_mb`.
  item = getattr(mod, name, None)
/Users/adrian/.conda/envs/lightning/lib/python3.8/site-packages/torch/jit/_recursive.py:701: LightningDeprecationWarning: The `LightningModule.model_size` property was deprecated in v1.5 and will be removed in v1.7. Please use the `pytorch_lightning.utilities.memory.get_model_size_mb`.
  item = getattr(nn_module, name, None)
/Users/adrian/.conda/envs/lightning/lib/python3.8/site-packages/torch/jit/_recursive.py:501: LightningDeprecationWarning: The `LightningModule.model_size` property was deprecated in v1.5 and will be removed in v1.7. Please use the `pytorch_lightning.utilities.memory.get_model_size_mb`.
  item = getattr(nn_module, name, None)
/Users/adrian/.conda/envs/lightning/lib/python3.8/site-packages/torch/jit/_recursive.py:584: LightningDeprecationWarning: The `LightningModule.model_size` property was deprecated in v1.5 and will be removed in v1.7. Please use the `pytorch_lightning.utilities.memory.get_model_size_mb`.
  item = getattr(nn_module, name, None)

```

### Expected behavior

No deprecation warning, since user did not have these properties in their code.

### Environment

```
* CUDA:
        - GPU:
        - available:         False
        - version:           None
* Packages:
        - numpy:             1.20.3
        - pyTorch_debug:     False
        - pyTorch_version:   1.10.0
        - pytorch-lightning: 1.5.0rc1
        - tqdm:              4.62.3
* System:
        - OS:                Darwin
        - architecture:
                - 64bit
                - 
        - processor:         i386
        - python:            3.8.11
        - version:           Darwin Kernel Version 19.6.0: Tue Jun 22 19:49:55 PDT 2021; root:xnu-6153.141.35~1/RELEASE_X86_64

```

cc @justusschock @awaelchli @akihironitta @rohitgr7 @tchaton

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes #10092 

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -1908,6 +1910,8 @@ def to_torchscript(
         """
         mode = self.training
 
+        self._running_torchscript = True
```
Comment (ananthsub) [inline pytorch_lightning/core/lightning.py:1913]: this should be reset at the end of the method

Comment (carmocca) [reply pytorch_lightning/core/lightning.py:1913]: Done!

Regarding the annotation, mypy does not require it.

**Review by akihironitta — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by awaelchli — APPROVED** [review]
We should keep the linked issue open because imo this is just a workaround for an issue in PyTorch.

**Comment by carmocca** [comment]
> We should keep the linked issue open because imo this is just a workaround for an issue in PyTorch.

You still have the issue in the PyTorch repository.

And we will likely remove the deprecation before they fix this.
